lib/Backends/Habana/Habana.cpp

```diff
@@ -504,7 +504,7 @@ llvm::Error HabanaFunction::execute(ExecutionContext *context) {
     EnqueueTensorInfo eti;
     llvm::StringRef name = P->getName();
     eti.tensorName = name.data();
-    eti.tensorSize = T->getSizeInBytes();
+    eti.tensorSize = T->getUnpaddedSizeInBytes();
```

Comment:
have you tested this :)?

Reply:
not yet, but since there are no virtually padded Tensors yet, unpaddedSizeInBytes = SizeInBytes.

Reply:
Can you test this in this PR via unit tests?

Reply:
good point, I'll add a unit test.